public/app/core/components/NavBar/NavBarMenu.tsx

```diff
@@ -359,33 +385,45 @@ function CollapsibleNavItem({
   onClose: () => void;
 }) {
   const styles = useStyles2(getCollapsibleStyles);
-  const [sectionExpanded, setSectionExpanded] = useLocalStorage(`grafana.navigation.expanded[${link.text}]`, false);
+  const sectionExpanded = link.expanded;
   const FeatureHighlightWrapper = link.highlightText ? NavFeatureHighlight : React.Fragment;
+  const isRoot = !link.parentItem;
+  const dispatch = useDispatch();
+
+  const handleToggle = (isOpen: boolean) => {
+    if (link.id) {
+      dispatch(updateMenuTree({ id: link.id, active: isOpen }));
+    }
+  };
 
   return (
-    <li className={cx(styles.menuItem, className)}>
-      <NavBarItemWithoutMenu
-        isActive={isActive}
-        label={link.text}
-        url={link.url}
-        target={link.target}
-        onClick={() => {
-          link.onClick?.();
-          onClose();
-        }}
-        className={styles.collapsibleMenuItem}
-        elClassName={styles.collapsibleIcon}
-      >
-        <FeatureHighlightWrapper>{getLinkIcon(link)}</FeatureHighlightWrapper>
-      </NavBarItemWithoutMenu>
+    <li className={cx(styles.menuItem, isRoot && styles.rootMenuItem, className)}>
+      {isRoot && (
+        <NavBarItemWithoutMenu
+          isActive={isActive}
+          label={link.text}
+          url={link.url}
+          target={link.target}
+          onClick={() => {
+            link.onClick?.();
+            onClose();
+          }}
+          className={cx(styles.collapsibleMenuItem, isRoot && styles.rootCollabsibleMenuItem)}
+          elClassName={styles.rootCollapsibleIcon}
+        >
+          <FeatureHighlightWrapper>{getLinkIcon(link, isRoot ? 'xl' : 'sm')}</FeatureHighlightWrapper>
+        </NavBarItemWithoutMenu>
+      )}
       <div className={styles.collapsibleSectionWrapper}>
         <CollapsableSection
+          controlled
           isOpen={Boolean(sectionExpanded)}
-          onToggle={(isOpen) => setSectionExpanded(isOpen)}
-          className={styles.collapseWrapper}
-          contentClassName={styles.collapseContent}
+          onToggle={handleToggle}
+          className={cx(styles.collapseWrapper, isRoot && styles.rootCollapseWrapper)}
+          contentClassName={cx(styles.collapseContent, isRoot && styles.rootCollapseContent)}
           label={
             <div className={cx(styles.labelWrapper, { [styles.primary]: isActive })}>
+              {!isRoot && <span className={styles.collapsibleIcon}>{getLinkIcon(link, isRoot ? 'xl' : 'sm')}</span>}
```

Comment:
the span will only show if `!isRoot`, so I believe that `isRoot ? 'xl' : 'sm'` condition is redundant